dcs-packaging-tool/dcs-packaging-tool-gui/src/main/java/org/dataconservancy/packaging/gui/presenter/impl/PackageMetadataPresenterImpl.java

```diff
@@ -0,0 +1,222 @@
+                    }
+                }
+            }
+        }
+    }
+
+    private void bind() {
+
+        // FIXME: The profile names should come from an actual service
+        view.loadDomainProfileNames(Arrays.asList("Bag-It", "Custom Profile", "Custom Profile 2"));
+        view.getAddDomainProfileButton().setOnAction(event -> {
+            if (view.getDomainProfilesComboBox().getSelectionModel().getSelectedItem() != null &&
+                    !view.getDomainProfilesComboBox().getSelectionModel().getSelectedItem().isEmpty()) {
+                view.addDomainProfileRemovableLabel(view.getDomainProfilesComboBox().getSelectionModel().getSelectedItem());
+            }
+        });
+
+        if (!view.isFormAlreadyDrawn()) {
+            view.setupRequiredFields(packageMetadataService.getRequiredPackageMetadata());
+            view.setupRecommendedFields(packageMetadataService.getRecommendedPackageMetadata());
+            view.setupOptionalFields(packageMetadataService.getOptionalPackageMetadata());
+        }
+
+        view.getContinueButton().setOnAction(event -> {
+            updatePackageState();
+
+            if (validateRequiredFields()) {
+                view.getErrorLabel().setVisible(false);
+                getController().goToNextPage();
+            } else {
+                view.getErrorLabel().setText(errors.get(ErrorKey.MISSING_REQUIRED_FIELDS));
+                view.getErrorLabel().setVisible(true);
+            }
+
+        });
+
+        view.getSaveButton().setOnAction(event -> {
+            updatePackageState();
+            packageMetadataFile = controller.showSaveFileDialog(view.getPackageMetadataFileChooser());
+            // TODO: Store the package metadata in some sort of file here.
+        });
+
+        view.getCancelLink().setOnAction(event -> view.showWarningPopup());
+
+        if (Platform.isFxApplicationThread()) {
+            view.getWarningPopup().setCancelEventHandler(event -> view.getWarningPopup().hide());
+            view.getWarningPopup().setConfirmEventHandler(event -> {
+                view.getWarningPopup().hide();
+                getController().goToPreviousPage();
+            });
+        }
+
+    }
+
+    /**
+     * helper method that updates the PackageState in the controller with the values in the form.
+     */
+    private void updatePackageState() {
+
+        getController().getPackageState().setPackageName(view.getPackageNameField().getText());
+
+        // Clear the package metadata list and reset the values based on the current state of form
+        getController().getPackageState().setPackageMetadataList(new LinkedHashMap<>());
+
+        // FIXME: This ID should come from somewhere? Setting something now so the page moves on.
+        getController().getPackageState().addPackageMetadata("BagIt-Profile-Identifier", "SomeFakeId");
+
+        for (Node removableLabel : view.getDomainProfileRemovableLabelVBox().getChildren()) {
```

Comment:
These fields are static AND required on the page, so we know they always exist and therefore easier to have explicit accessors for them to fetch their values. I will put this comment in the code.